tests/unit/fixtures/database.py

```diff
@@ -0,0 +1,172 @@
+"""
+Fixtures for setting up and managing the database and test environment for unit tests.
+
+This module provides a set of pytest fixtures for initializing and interacting with the test database,
+resetting the database schema between tests, and populating the database with initial data for testing.
+The fixtures also include necessary session management and integration with SQLAlchemy.
+"""
+
+import os
+import typing as t
+from datetime import datetime
+
+import pytest
+import sqlalchemy as sa
+from sqlalchemy.orm import Session, sessionmaker
+
+from casp.services import settings
+from casp.services.api import schemas
+from casp.services.db import Base, create_engine, models
+from tests.unit.fixtures import constants
+
+
+@pytest.fixture(scope="session", autouse=True)
+def test_engine() -> t.Generator[sa.engine.Engine, None, None]:
+    """
+    Fixture to create a test database engine. Create all tables before tests and remove tables and test database file
+    after
+
+    **Scope**: Session
+
+    **Autouse**: Yes
+
+    :return: A SQLAlchemy engine connected to the test database.
+    """
+    # Execute before all tests
+    engine = create_engine()
+    Base.metadata.create_all(engine)
+    yield engine  # Let the tests run
+    # Execute after all tests
+    Base.metadata.drop_all(engine)
+
+    if os.path.exists(settings.TEST_DB_FILE_PATH):
+        os.remove(settings.TEST_DB_FILE_PATH)
+
+
+@pytest.fixture
+def db_session(test_engine: sa.engine.Engine) -> t.Generator[Session, None, None]:
+    """
+    Fixture to provide a SQLAlchemy session for querying during tests.
+
+    **Scope**: Function
+
+    :param test_engine: The SQLAlchemy engine used for the test database.
+
+    :return: A generator that yields a SQLAlchemy session.
+    """
+    session_local_class = sessionmaker(bind=test_engine)
+    with session_local_class() as session:
+        yield session
+
+
+@pytest.fixture(scope="function", autouse=True)
+def reset_db(test_engine: sa.engine.Engine) -> t.Generator[None, None, None]:
```

Comment:
does this happen after every pytest test function run?  If so I'm curious how much time is being added to your full unit test suite.

Reply:
Update: now after I updated this from executing after each of the test, to only the tests where the db is used it's 3-4 seconds.